cf-smoke/src/main/java/com/adobe/cq/cloud/testing/it/cf/smoke/rules/InstallPackageRule.java

```diff
@@ -117,17 +115,25 @@ public void evaluate() throws Throwable {
     }
 
     private File generatePackage(String resourceFolder) throws IOException, URISyntaxException {
-        URI resourceUri = getClass().getResource(resourceFolder).toURI();
+        URI resourceUri = Objects.requireNonNull(getClass().getResource(resourceFolder)).toURI();
         Path resourcePath;
-        if (resourceUri.getScheme().equals("jar")) {
-            FileSystem fs = FileSystems.newFileSystem(resourceUri, Collections.emptyMap());
-            resourcePath = fs.getPath(resourceFolder);
-        } else {
-            resourcePath = Paths.get(resourceUri);
-        }
 
-        LOG.info("Creating package from resources folder {}", resourcePath);
-        return buildJarFromFolder(resourcePath);
+        FileSystem fs = null;
+        try {
+            if (resourceUri.getScheme().equals("jar")) {
+                fs = FileSystems.newFileSystem(resourceUri, Collections.emptyMap());
+                resourcePath = fs.getPath(resourceFolder);
+            } else {
+                resourcePath = Paths.get(resourceUri);
+            }
+
+            LOG.info("Creating package from resources folder {}", resourcePath);
+            return buildJarFromFolder(resourcePath);
+        } finally {
+            if (fs != null) {
+                fs.close();
```

Comment:
Since we are sprinkling try-with-resources everywhere, looks like it could be done here?